### DISCUSSION TITLE
Google, Android 13 and G6

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2280] [Q&A]
Hi Pixel users,
the Android 13 rollout has started, which is very good news given the new features.  My question to you would now be whether there are already problems with xDrip and the Dexcom G6 with Android 13 and the Pixel smartphones?  I'm considering switching to the upcoming Pixel 7 (Pro) as I'm having Bluetooth connectivity issues with my current device.  Has anyone had any experience with the Teramisu update?

**Comment by meierthomas** [comment]
I'm in the same boat. When I got my Pixel 6 Pro I realised xDrip didn't work so I installed it on my Pixel 4 XL. Now I would to switch back to my Pixel 6 Pro in the interim until the Pixel 7 Pro is launched running Android 13.

Does anyone use xDrip/xDrip+ on a Pixel 6 Pro with Android 12 or 13?

**Reply by jamorham** [reply]
G6 on Pixel 4 with Android 13 appears to be working.

**Reply by whoiswes** [reply]
G6 on Pixel 4a with Android 13 working fine here.

**Reply by Miquest** [reply]
Thank you for the answers!
My Telekom contract has to be renewed in December, I will then look for a Pixel 7 Pro. I hope it works or someone has a Pixel 7 Pro at hand from October and can confirm the functionality with the Dexcom G6. My reason for getting this smartphone is that the 7Pro will probably be compatible with the T: Connect Mobile Application. I will see if it works....

**Reply by meierthomas** [reply]
I'll be getting the Pixel 7 Pro as well. With the launch in October I'm not sure about availability in Australia but will let you know. At the moment I'm using xDrip+ on my Pixel 4 XL as the last time I checked it didn't work on the Pixel 6 Pro running Android 12.

**Comment by bryanw20** [comment]
Pixel 6a, and 7pro, works 99% of the time.
I do occasionally get a period of 30-45min of missed signal.
Sometimes it starts on its own, sometimes a reboot fixed it.

**Reply by Navid200** [reply]
Have you set the Bluetooth battery setting (Android, not xDrip) to unrestricted?

**Reply by bryanw20** [reply]
I have the xdrip app set to unrestricted in adaptive battery settings

**Reply by Navid200** [reply]
That's good.  But, that was not my question.

Look at this under the heading "Bluetooth Battery Optimization", which comes after the heading "Battery Optimization".
https://navid200.github.io/xDrip/docs/Dexcom-Basics

**Reply by Miquest** [reply]
The Bluetooth Battery optimization does not exist in Android 13!
Also in Android 12 and I think also 11.

**Reply by bryanw20** [reply]
Ok, think I found it, same setting as for xdrip app, but for bt. Set it to unrestricted.

We'll see if that changes anything.

Like I said mines maybe once a week that I noticed it.

**Reply by bryanw20** [reply]
Got it again last night, idk if coincidence but about the same time I turned on by headphones.

![Screenshot_20221212-001758](https://user-images.githubusercontent.com/26827860/207080887-3f042646-d0b6-4230-b561-455400707ed0.png)

**Reply by Navid200** [reply]
Nothing in those logs suggests a disconnect.

**Reply by bryanw20** [reply]
I know, that's part of what's weird. Anything I can send to help?

**Comment by bryanw20** [comment]
![Screenshot_20221210-093402](https://user-images.githubusercontent.com/26827860/206862909-2f083242-482b-4eb1-88a2-b165f97e9bc5.png)

**Reply by Miquest** [reply]
Oh, at the Pixel it does exist...

**Comment by predic8** [comment]
Hi @bryanw20,
you wrote that it works von Pixel 7 Pro. How could you install it? I get an error that the app is too old and that for security reasons the app could not installed. Even if I trust the source.

**Reply by AphgaDesign** [reply]
I have the same issue, the app never install

**Reply by Navid200** [reply]
Please have a look at the guide here: https://navid200.github.io/xDrip/docs/Install.html

The app is too old is a common error message.  Just ignore it.

**Reply by AphgaDesign** [reply]
I did that like this
![Screenshot_20230501-221938.png](https://user-images.githubusercontent.com/17551643/235565785-e52b58db-5e8d-4d14-90f7-ae832bd6161c.png)

![Screenshot_20230501-221948.png](https://user-images.githubusercontent.com/17551643/235565803-bb76b728-b513-4418-9431-696fabf5cfa3.png)

![Screenshot_20230501-221957.png](https://user-images.githubusercontent.com/17551643/235565811-a117b964-996b-4da7-92f2-b670d52f727d.png)

And still not installing

**Reply by Navid200** [reply]
Did you tap on "Plus de details"?

**Reply by AphgaDesign** [reply]
Yes, here is the message

![Screenshot_20230501-232928.png](https://user-images.githubusercontent.com/17551643/235573669-2331db78-d7a9-4870-b120-19d98be9f374.png)

**Reply by Miquest** [reply]
If you tap on "Installer quand même", it should Work

**Reply by AphgaDesign** [reply]
Yes, thanks

**Reply by Miquest** [reply]
You're welcome :)

**Reply by Navid200** [reply]
But, that's exactly what is described in the guide,

It would be greatly appreciated if you tell me how I can edit the guide so that it is more clear.

**Reply by AphgaDesign** [reply]
Just not clear that "Installer quand même" is clickable

**Reply by Navid200** [reply]
Thanks.  
I have added a paragraph at the bottom of the page hoping to avoid the confusion.

**Reply by tubayer** [reply]
Hi, 
I could install xdrips with my Pixel 7 Pro on the newest Android. What helped was checking out the project from git, opening it in Android Studio and syncing it onto my phone via USB and the developer settings on the Pixel. Sounds complicated but it took only about 10 minutes.

**Reply by Navid200** [reply]
@tubayer 
The purpose of these discussions is to help xDrip users.
There is absolutely no need to use Android Studio in order to install xDrip.  Stating that here does not help anyone.  It could only confuse people.
